nmagent/responses.go

```diff
@@ -38,7 +38,8 @@ type NCVersionList struct {
 }
 
 type AzResponse struct {
-	HomeAz uint `json:"homeAz"`
+	HomeAz     uint `json:"homeAz"`
+	APIVersion uint `json:"apiVersion"`
```

Comment:
How is this used? Nothing else seems to change, so why does anything care what APIVersion is present?

Reply:
This is going to be consumed in dnc and used for indicating a Dualstack fix from NMA

Reply:
Can you describe the ipv6 fix needed in a little more detail? My concern with this is that nothing about the API is actually changing. We're just adding a cryptic field to it that clients are just supposed to understand. What does it mean when `APIVersion` becomes 3, 4, so on? A more obvious field would be something like `NeedsIPv6Fix = true` or something, but I don't have enough context on the entire problem to recommend a good API (hence the request for more detail).

Reply:
Currently NMA is rollout a new fix for the azr swift regarding dualstacks secnarios. they are adding a new field in home az   api to indicate the fix already there, so dnc will check this apiVersion coming from NMA response to decide enable azr or not. This APIVersion will always be 2. we should raise alert if we saw apiVersion other than 2.

Reply:
but NMA mentioned this value might change in the future for different purpose

Reply:
If NMAgent is trying to indicate whether or not AZR should be enabled based on some programming they do, then the field should be `EnableAZR = true / false`. We shouldn't care about some fix or whatever that they have to do. Can we push back on this? The fact of NMAgent's APIs being generally not great isn't justification for allowing them to continue to make them worse.

If the answer is "no, the decision is firm," then we should fix this at the system border so we don't have to spread the knowledge about what a `2` means through the entire ACN stack. The NMAgent client (whose designed responsibility is a respository of all of the weird things NMAgent does) should translate this to something meaningful like `EnableAZR` in the response.

Reply:
While I agree with your comment, Tim, NMA wants us to really use APIVersion here where they can probably change version to reuse this in case, they need to provide such a fix for another issue later (just having a boolean for one case might warrant more changes later from their side). That being said, @ZetaoZhuang we should check with NMA if 0 and 2 are the only values we should expect from them.

I agree this is not clean at all, but this is what they're pushing us to use from their side. Currently, this is a hotfix like change for us and ideally, we shouldn't need to use this later.

Reply:
I checked with Vishwesh, he confirmed that 2 is the only valid value for now. we should also allow 0 as well. that is the default value of the field when NMA is still running the old version, and apiVersion does not exist

Reply:
Okay, thanks for clarifying that we don't have any control over what they do. I'll make suggestions to isolate their bad design, though, so we don't have to deal with it forever.